### PR TITLE
Allow custom renderer for iso

### LINKED ISSUE #9: layout consequences of wrapping the application in a div
Might be worth documenting the HTML that iso generates because it looks like I'll need to style it.

For example, I have a flexbox-based layout that expects `html` and `body` to be 100% high. The app is then grown to fill that space. But since iso wraps the html – `div.___iso-html___` – the layout breaks…unless that iso wrapper is styled to be 100% high too.

Or, iso could allow an extra class to be set so that things don't break if that classname is ever altered in this module.

What do you think?

### PR DESCRIPTION
**Pull request by goatslacker**
- Adds much needed tests to this project
- Adds some linting

Breaking change for iso.

You can now specify your own custom renderer. By default it'll render the markup in a div and the state in a script with application/json type. You can then also pass in your own selector function which will return the nodes and its corresponding state pairs.

Pros:
- More customizable.
- Less code.

TODO
- [x] Docs
- [x] Changelog

to @ljharb @lelandrichardson @michaelBenin 

Fixes #9

### CODE REVIEW

**Comment by ljharb** [comment]
It seems like the linting change could go in as a non-breaking separately from the custom renderer?

File: src/iso.js [hunk 1]
```diff
@@ -1,99 +1,77 @@
...
+const defaultSelector = () => {
+  const all = document.querySelectorAll(`[${KEY_NAME}]`)
+
+  return Array.prototype.reduce.call(all, (cache, node) => {
+    const key = node.getAttribute(KEY_NAME)
+
+    cache[key] = cache[key] || {}
+
+    if (node.nodeName === 'SCRIPT') {
+      const state = JSON.parse(node.innerHTML)
```
Comment (michaelBenin) [inline src/iso.js]: Might be more defensive to wrap in a try catch?

Comment (goatslacker) [inline src/iso.js]: Good call.

**Comment by ljharb** [comment]
LGTM

**Pull request merged**